### ISSUE TITLE
Allow LABELs to accept array syntax, for values that should be exec-ed without shell

### ISSUE THREAD

**Issue by adelton** [docker/docker#21060]
When `LABEL`s are used to store commands, for example ways to run the container, it might be useful to be able to pass the command and list of parameters directly to `exec`. However `LABEL` values are currently strings only, so parsing of that string is needed, either by splitting on space or passing it to shell.

It'd be useful to be able to specify the `LABEL` value as array, and have it consumed like that.

For example

```
LABEL INSTALL [ 'docker', 'run', '--rm', '${NAME}', '/bin/test.sh', 'install', 'test install with space, as well as comma' ]
```

Output of `docker version`:

```
Client:
 Version:         1.9.1
 API version:     1.21
 Package version: docker-1.9.1-6.git6ec29ef.fc23.x86_64
 Go version:      go1.5.3
 Git commit:      7206621
 Built:           Mon Jan 25 20:52:22 UTC 2016
 OS/Arch:         linux/amd64

Server:
 Version:         1.9.1
 API version:     1.21
 Package version: docker-1.9.1-6.git6ec29ef.fc23.x86_64
 Go version:      go1.5.3
 Git commit:      7206621
 Built:           Mon Jan 25 20:52:22 UTC 2016
 OS/Arch:         linux/amd64
```

Output of `docker info`:

```
Containers: 1
Images: 40
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker-253:0-67643278-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 1.015 GB
 Data Space Total: 107.4 GB
 Data Space Available: 51.26 GB
 Metadata Space Used: 2.736 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.109 (2015-09-22)
Execution Driver: native-0.2
Logging Driver: journald
Kernel Version: 4.4.3-300.fc23.x86_64
Operating System: Fedora 23 (Twenty Three)
CPUs: 4
Total Memory: 7.795 GiB
Name: docker.example.com
ID: 7P7U:L4GE:CMFK:RSXI:NPSX:UVBK:VO5U:3K2W:LLMG:I4UJ:Q4LX:622P
```

Provide additional environment details (AWS, VirtualBox, physical, etc.):

List the steps to reproduce the issue:
1. Use `Dockerfile` with

```
LABEL INSTALL [ 'docker', 'run', '--rm', '${NAME}', '/bin/test.sh', 'install', 'test install with space, as well as comma' ]
LABEL RUN 'docker run --rm ${NAME} /bin/test.sh run test run'
CMD [ "/bin/test.sh", "cmd", "cmd with space" ]
```

2. Built the image, inspect it.

Describe the results you received:

```
    "ContainerConfig": {
        ...
	"Cmd": [
            "/bin/sh",
            "-c",
            "#(nop) CMD [\"/bin/test.sh\" \"cmd\" \"cmd with space\"]"
        ],
	...
	"Labels": {
            "INSTALL": "[ docker, run, --rm, ${NAME}, /bin/test.sh, install, test install with space, as well as comma ]",
            "RUN": "docker run --rm ${NAME} /bin/test.sh run test run"
        }
    },
    ...
    "Config": {
        ...
	"Cmd": [
            "/bin/test.sh",
            "cmd",
            "cmd with space"
        ],
	...
	"Labels": {
            "INSTALL": "[ docker, run, --rm, ${NAME}, /bin/test.sh, install, test install with space, as well as comma ]",
            "RUN": "docker run --rm ${NAME} /bin/test.sh run test run"
        }
    },
```

Describe the results you expected:

```
    "ContainerConfig": {
        ...
	"Cmd": [
            "/bin/sh",
            "-c",
            "#(nop) CMD [\"/bin/test.sh\" \"cmd\" \"cmd with space\"]"
        ],
	...
	"Labels": {
            "INSTALL": [
                  "docker", "run", "--rm", "${NAME}", "/bin/test.sh", "install", "test install with space, as well as comma"
                  ],
            "RUN": "docker run --rm ${NAME} /bin/test.sh run test run"
        }
    },
    ...
    "Config": {
        ...
	"Cmd": [
            "/bin/test.sh",
            "cmd",
            "cmd with space"
        ],
	...
	"Labels": {
            "INSTALL": [
                  "docker", "run", "--rm", "${NAME}", "/bin/test.sh", "install", "test install with space, as well as comma"
                  ],
            "RUN": "docker run --rm ${NAME} /bin/test.sh run test run"
        }
    },
```

Basically when the `LABEL` value is specified with the array syntax, I'd expect the result to be similar to the one for `CMD`.

Provide additional info you think is important:

**Comment by thaJeztah** [comment]
Thanks for suggesting, @adelton, but this is not something we will implement. When designing the "labels" feature, we deliberately kept them simple and only accept a string. Docker is treating label values as an opaque string, and won't do any parsing/interpreting of its value. Adding this feature would add a lot of complexity (think, for example of `docker ps --filter label=foo=bar`; what should happen if "foo" is an array?), that we don't want.

I'm sorry, but I'll close this issue, because this is not something we'll implement.

Thanks for opening this issue though, it's appreciated!
